### PR TITLE
Update extractors

### PR DESCRIPTION
**Pull request by Lalaland**
This PR does two things:

1. It refactors the extractors to match a more spread out design.

2. It fixes various bugs and issues that were preventing a full extract on all of STARR OMOP

CC @rthapa84

### CODE REVIEW

**Comment by Lalaland** [comment]
Closes #14

File: src/piton/extractors/csv.py [hunk 1]
```diff
@@ -140,48 +140,48 @@ def run_csv_converters(
...
             ]
-            if len(matching_converters) > 1:
+            if len(matching_extractors) > 1:
                 print(
                     "Multiple converters matched?",
                     full_path,
-                    matching_converters,
+                    matching_extractors,
                 )
                 print(1 / 0)
```
Comment (rthapa84) [inline src/piton/extractors/csv.py:165]: Why do we want to throw division by zero error here?

File: src/piton/extractors/csv.py [hunk 2]
```diff
@@ -140,48 +140,48 @@ def run_csv_converters(
...
-            elif len(matching_converters) == 0:
+            elif len(matching_extractors) == 0:
```
Comment (rthapa84) [inline src/piton/extractors/csv.py:166]: For consistency, we probably want to have print("No converters matched", full_path, matching_extractors) after this line?

File: src/piton/extractors/csv.py (hunk 1 above)
Comment (Lalaland) [reply src/piton/extractors/csv.py:165]: Oh oops, this should be replaced by an actual exception. print (1/0) is just a hack to generate an easy to debug exception in a small amount of code.

File: src/piton/extractors/csv.py (hunk 2 above)
Comment (Lalaland) [reply src/piton/extractors/csv.py:166]: Yeah, this should throw an exception as well with context.

File: src/piton/extractors/omop.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+                month = int(row["month_of_birth"])
+
+            if row["day_of_birth"]:
+                day = int(row["day_of_birth"])
+
+            birth = datetime.datetime(year=year, month=month, day=day)
+
+        return [
+            # 4216316 is the OMOP birth code
+            Event(start=birth, code=4216316, event_type=row["load_table_id"])
```
Comment (rthapa84) [inline src/piton/extractors/omop.py:49]: Maybe I am a bit confused here, but when I look up for birth on Athena Ohdsi, birth has Concept ID of 4216316 and concept code of 3950001. Here, we are noting it as code.

File: src/piton/extractors/omop.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            )
+        ]
+
+
+def get_omop_csv_extractors() -> Sequence[CSVExtractor]:
+    """Get the list of OMOP Converters."""
+    converters = [
+        _DemographicsConverter(),
+        _ConceptTableConverter(
+            "drug_exposure",
```
Comment (rthapa84) [inline src/piton/extractors/omop.py:167]: Let's name these as prefix for consistency and code clarity.

File: src/piton/extractors/omop.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    ) -> Optional[datetime.datetime]:
+        """Extract the highest resolution date from the raw data."""
+        for attempt in (date_field + "time", date_field):
+            if attempt in row and row[attempt] != "":
+                return datetime.datetime.fromisoformat(row[attempt])
+
+        return None
+
+    def get_events(self, row: Mapping[str, str]) -> Sequence[Event]:
+        def normalize_to_float_if_possible(
```
Comment (rthapa84) [inline src/piton/extractors/omop.py:105]: It is a bit weird that we have this function inside get_events function inside the class but we have the function _try_numeric that this function is dependent on entirely outside the class.

Comment (Lalaland) [reply src/piton/extractors/omop.py:105]: _try_numeric is generic and doesn't depend on the exact logic of this class. normalize_to_float_if_possible does depend entirely on the internal logic here.

File: src/piton/extractors/omop.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+from __future__ import annotations
+
+import dataclasses
+import datetime
+from typing import Mapping, Optional, Sequence
+
+from piton import Event
+from piton.extractors.csv import CSVExtractor
+
+OMOP_BIRTH = 4216316
```
Comment (rthapa84) [inline src/piton/extractors/omop.py:12]: I see that it is being used in __init__.py file inside transforms, so why is it being defined in this file?

File: src/piton/transforms/__init__.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            new_events.append(dataclasses.replace(event, start=new_time))
+        else:
+            new_events.append(event)
+
+    new_events.sort(key=lambda a: (a.start, a.code))
+
+    return Patient(patient.patient_id, new_events)
+
+
+def delta_encode(patient: Patient) -> Patient:
```
Comment (rthapa84) [inline src/piton/transforms/__init__.py:105]: I may be completely misunderstanding it, but I was thinking that events can be anything like lab measurements or vital signs, which are can be taken multiple in a day and can be duplicates. Do we want to delete such duplicated?

File: src/piton/extractors/omop.py (hunk 6 above)
Comment (Lalaland) [reply src/piton/extractors/omop.py:12]: Fixed by moving the code to transforms/stanford.py

File: src/piton/transforms/__init__.py (hunk 7 above)
Comment (Lalaland) [reply src/piton/transforms/__init__.py:105]: Generally, yes, we probably want to delete those types of duplicates.

Comment (Lalaland) [reply src/piton/transforms/__init__.py:105]: Yeah, generally we want to remove those types of duplicates. We will have information loss either way and that's the best information to drop.

Comment (rthapa84) [reply src/piton/transforms/__init__.py:105]: Sounds good to me. Thank you for the clarification.

**Pull request merged**